package/upgrade/upgrade_manifests.sh

```diff
@@ -903,6 +903,56 @@ wait_for_fleet_agent(){
   done
 }
 
+upgrade_harvester_csi_rbac() {
+
+  # only versions before v1.4.0 that upgrading to v1.4.0 need this patch
+  if [[ ! "${UPGRADE_PREVIOUS_VERSION%%-rc*}" < "v1.4.0" ]]; then
+    echo "Only versions before v1.4.0 need this patch."
+    return
+  fi
+
+  if kubectl get clusterrole harvesterhci.io:csi-driver 2> /dev/null; then
+    echo "Upgrade ClusterRole harvesterhci.io:csi-driver ..."
+
+    cat <<EOF | kubectl apply -f -
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  labels:
+    app.kubernetes.io/component: apiserver
+    app.kubernetes.io/name: harvester
+    app.kubernetes.io/part-of: harvester
+  name: harvesterhci.io:csi-driver
+rules:
+- apiGroups:
+  - storage.k8s.io
+  resources:
+  - storageclasses
+  verbs:
+  - get
+  - list
+  - watch
+- apiGroups:
+  - harvesterhci.io
+  resources:
+  - networkfilesystems
+  - networkfilesystems/status
+  verbs:
+  - '*'
+- apiGroups:
+  - longhorn.io
+  resources:
+  - volumes
+  - volumes/status
+  verbs:
+  - get
+  - list
+EOF
+  else
+    echo "ClusterRole harvesterhci.io:csi-driver not found, skip updating."
```

Comment:
Would you mind explaining why we are skipping the update for the ClusterRole when it does not exist? I thought we need the ClusterRole for the CSI driver's RWX function to work correctly. Thank you.

Reply:
Hi @starbops, thanks for the review

Because only the user manually removes the ClusterRole, the ClusterRole will not exist.
And it only affects the downstream cluster. I don't want to block the upgrade in this situation.
For sure, we have a simple workaround for manually creating this ClusterRole.

That's why I thought we could skip here.

Reply:
That makes sense. Thanks for the explanation.